CONTRIBUTING.md

```diff
@@ -1,133 +1,438 @@
-# Contributing to Apache NuttX RTOS
+# Apache NuttX RTOS Contributing Guidelines
 
-Hi! Thank you for your interest in contributing to Apache NuttX RTOS :-)
+Thank you for your interest in contributing to Apache NuttX RTOS :-)
 
-## Guidelines
+If you haven't yet read
+[The Inviolable Principles of NuttX]( https://github.com/apache/nuttx/blob/master/INVIOLABLES.md)
+please do so right now.
 
-To help us successfully review your contribution, it is very
-important that you follow these guidelines.
+Please note that NuttX supports over 15 different architectures, 360+ boards,
+and 1600+ configurations, that are used in various project and products around
+the world. Remember that any code change may affect different users and their
+business. Please try to keep your contributions high quality and compatible.
 
-If you need more information please read the [Full Contributing Documentation](https://nuttx.apache.org/docs/latest/contributing/index.html).
+To help us smoothly process your contributions it is very important that you
+follow the guidelines. These rules are based on daily experiences and helps
+us preserve long term self-compatibility and maintenance of the project.
+NuttX is maintained by a small team of volunteers from around the world.
```

Comment:
This is a good sentence, but I think it might need to move to elsewhere in the document, because it doesn't quite fit with the flow of this part of the text.

Reply:
hmm, i used top-bottom approach.. did small rearrangement.. what place would be better or what better text? :-)